### ISSUE TITLE
Build problems

### ISSUE THREAD

**Issue by PaulDiraq** [google/benchmark#314]
I tried to build the project without libc++

and encountered the following issue:

While building with cmake (and without using libc++) all feature_checks failed with:

```
-- Performing Test HAVE_POSIX_REGEX
CMake Error at cmake/CXXFeatureCheck.cmake:28 (try_run):
  LINK_LIBRARIES specified but there is no content
Call Stack (most recent call first):
  CMakeLists.txt:89 (cxx_feature_check)
```

Problem: `cxx_feature_check` populates the `try_run` `LINK_LIBRARIES` parameter with  `${BENCHMARK_CXX_LIBRARIES}` (CXXFeatureCheck.cmake:31).

`BENCHMARK_CXX_LIBRARIES` is only set if` BENCHMARK_USE_LIBCXX `is `ON`. 
(CMakeLists.txt:151).

My workaround(I don't know anything about cmake):  
In file include CMakeLists.txt:35
`   list(APPEND BENCHMARK_CXX_LIBRARIES util)`
right after 
`   include(CXXFeatureCheck)`
to have BENCHMARK_CXX_LIBRARIES set (this will link against some libutil which hopefully has no used objects.)

Now working; all test are passing.

compiler
gcc (SUSE Linux) 4.8.1 20130909 [gcc-4_8-branch revision 202388]

cmake:
cmake version 2.8.11.2

OS:
openSUSE 13.1 (Bottle) (x86_64)

**Comment by EricWF** [comment]
This seems to be a bug/legacy behavior for CMake 2.8.11 which is changed/fixed in 2.8.12. [This commit](https://git.backbone.ws/cmake/cmake/commit/d06db7ebe80636876d9701064b16cec9d3e2e3cb) removes the behavior from CMake.

I think we should bump the CMake minimum version to 2.8.12.

**Comment by EricWF** [comment]
Closing since we bumped the minimum CMake version to 2.8.12, which no longer exhibits this problem.
